src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -156,183 +187,5337 @@ static ColliderCylinderInit D_8094A840 = {
+                sKotakePtr->actor.world.pos.y = 263.0f;
+                sKoumePtr->actor.world.pos.y = sKotakePtr->actor.world.pos.y;
+                this->subCamAt.y = sKoumePtr->actor.world.pos.y + 17.0f;
+                sKotakePtr->actor.world.pos.z = 0.0f;
+                sKoumePtr->actor.world.pos.z = sKotakePtr->actor.world.pos.z;
+                sKoumePtr->work[YAW_TGT] = sKotakePtr->work[YAW_TGT] = sKoumePtr->actor.shape.rot.x =
+                    sKotakePtr->actor.shape.rot.x = sKoumePtr->actor.shape.rot.y = sKotakePtr->actor.shape.rot.y = 0;
+                func_8002DF54(globalCtx, &sKoumePtr->actor, 1);
+                sKoumePtr->actor.flags |= 1;
+            }
+            break;
+        case 2:
+            if (this->work[CS_TIMER_2] == 100) {
+                Vec3f pos;
+                Vec3f velocity;
+                Vec3f accel = { 0.0f, 0.0f, 0.0f };
+                for (i = 0; i < 50; i++) {
+                    velocity.x = Rand_CenteredFloat(3.0f);
+                    velocity.y = Rand_CenteredFloat(3.0f);
+                    velocity.z = Rand_CenteredFloat(3.0f);
+                    pos = sKoumePtr->actor.world.pos;
+                    pos.x += velocity.x * 2.0f;
+                    pos.y += velocity.y * 2.0f;
+                    pos.z += velocity.z * 2.0f;
+                    BossTw_AddFlameEffect(globalCtx, &pos, &velocity, &accel, Rand_ZeroFloat(2.0f) + 5, 1);
+                    velocity.x = Rand_CenteredFloat(3.0f);
+                    velocity.y = Rand_CenteredFloat(3.0f);
+                    velocity.z = Rand_CenteredFloat(3.0f);
+                    pos = sKotakePtr->actor.world.pos;
+                    pos.x += velocity.x * 2.0f;
+                    pos.y += velocity.y * 2.0f;
+                    pos.z += velocity.z * 2.0f;
+                    BossTw_AddFlameEffect(globalCtx, &pos, &velocity, &accel, Rand_ZeroFloat(2.0f) + 5, 0);
+                }
+                Actor_SetScale(&sKoumePtr->actor, 0.0f);
+                Actor_SetScale(&sKotakePtr->actor, 0.0f);
+                sKoumePtr->visible = 1;
+                sKotakePtr->visible = 1;
+                func_80078884(NA_SE_EN_TWINROBA_TRANSFORM);
+                Audio_QueueSeqCmd(0x61);
+                this->csState2 = 3;
+                this->work[CS_TIMER_2] = 0;
+                this->subCamYaw = this->subCamYawStep = this->actor.speedXZ = this->subCamDistStep = 0.0f;
+            }
+            break;
+        case 3:
+            BossTw_DeathCSMsgSfx(this, globalCtx);
+            if (this->work[CS_TIMER_2] < 150) {
+                globalCtx->envCtx.unk_BE = 1;
+                globalCtx->envCtx.unk_BD = 0;
+                Math_ApproachZeroF(&globalCtx->envCtx.unk_D8, 1.0f, 0.1f);
+            } else {
+                globalCtx->envCtx.unk_BE = 1;
+                globalCtx->envCtx.unk_BD = 6;
+                Math_ApproachF(&globalCtx->envCtx.unk_D8, (Math_SinS(this->work[CS_TIMER_2] * 4096) / 4.0f) + 0.75f,
```

Comment:
```suggestion
                Math_ApproachF(&globalCtx->envCtx.unk_D8, (Math_SinS(this->work[CS_TIMER_2] * 0x1000) / 4.0f) + 0.75f,
```
It is an angle, after all.